enterprise/core-edge/src/main/java/org/neo4j/coreedge/raft/replication/tx/ExpontentialBackoffStrategy.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (c) 2002-2016 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.coreedge.raft.replication.tx;
+
+import java.util.concurrent.TimeUnit;
+
+public class ExpontentialBackoffStrategy implements RetryStrategy
+{
+    private long timeout;
+    private long previous;
+
+    public ExpontentialBackoffStrategy( long initialTimeout, TimeUnit timeUnit )
+    {
+        this.timeout = timeUnit.toMillis( initialTimeout );
+    }
+
+    @Override
+    public long nextTimeout()
+    {
+        previous = timeout;
```

Comment:
It would be nice if there was a cap here, otherwise meaningless high values will be reached.